### PR TITLE
Renamed EventsBase to EventSetBase (and derivatives)

### PR DESCRIPTION
**Pull request by facundobatista**
This is per what we discussed in Frankfurt, as EventsBase was too confusing with EventBase, and we agreed on Signal to name it EventSetBase.

Also for the files I changed I updated copyright years, and removed execution infrastructure of the test (as we're using proper test runners).

### CODE REVIEW

File: test/test_framework.py [hunk 1]
```diff
@@ -22,8 +21,20 @@
...
+    EventSource,
+    Framework,
+    Handle,
+    NoSnapshotError,
+    Object,
+    PreCommitEvent,
+    SQLiteStorage,
+    StoredList,
+    StoredState,
+    StoredStateData,
```
Comment (chipaca) [inline test/test_framework.py]: :heartbeat:

**Comment by chipaca** [comment]
@jameinel hm, I think we should also rename the remaining three, don't you?
That is: `CharmEvents`, `PrefixedEvents`, and `StoredStateEvents`.

**Comment by jameinel** [comment]
CharmEvents honestly reads quite nicely. I'm not opposed to calling it
ChamEventSet, but CharmEvents does read better to me.

**Comment by chipaca** [comment]
that's a good point, there isn't the tension there was with `EventBase` vs `EventsBase`; `FooEvents` is probably fine.
OK.

**Pull request merged**